azure-ipam/ipam_test.go

```diff
@@ -85,6 +88,114 @@ func (c *MockCNSClient) RequestIPAddress(ctx context.Context, ipconfig cns.IPCon
 	}
 }
 
+func (c *MockCNSClient) RequestIPs(ctx context.Context, ipconfig cns.IPConfigsRequest) (*cns.IPConfigsResponse, error) {
+	switch ipconfig.InfraContainerID {
+	case "failRequestCNSArgs":
+		return nil, errFoo
+	case "happyArgsSingle", "failProcessCNSRespSingleIP", "failRequestCNSArgsSingleIP":
+		e := &client.CNSClientError{}
+		e.Code = types.UnsupportedAPI
+		e.Err = errUnsupportedAPI
+		return nil, e
```

Comment:
@tamilmani1989 For `RequestIPs` each of these tests will return back the UnsupportedAPI when we call `RequestIPs` which then causes us to test `RequestIPAddress`